frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,356 @@
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('pkg')}`,
+    selector: {
+      matchLabels: {
+        catalog: new URLSearchParams(window.location.search).get('catalog'),
+      },
+    },
+    prop: 'packageManifest',
+  },
+];
+
+const OperatorInstallPageWithoutFirehose: React.FC<OperatorInstallPageProps> = (props) => {
+  const [loading, setLoading] = React.useState(true);
+  const [installObj, setInstallObj] = React.useState<ClusterServiceVersionKind | InstallPlanKind>();
+  const [status, setStatus] = React.useState('');
+  const [needsApproval, setNeedsApproval] = React.useState(false);
+  const [packageManifest, setPackageManifest] = React.useState<PackageManifestKind>();
+
+  const { resources, namespace } = props;
+  const csvName = new URLSearchParams(window.location.search).get('csvName');
+
+  const isStatusSucceeded = () => {
+    return status === 'Succeeded' && _.get(installObj, 'metadata.namespace') === namespace;
+  };
+  const isStatusFailed = () => {
+    return status === 'Failed';
+  };
+  const isApprovalNeeded = () => {
+    return needsApproval;
+  };
+
+  React.useEffect(() => {
+    if ([...(resources.packageManifest.data as PackageManifestKind[])].length > 0) {
+      setPackageManifest(resources.packageManifest.data[0]);
+    }
+    const tmpCSV = [
+      ...(resources.clusterServiceVersions.data as ClusterServiceVersionKind[]),
+      ...(resources.allClusterServiceVersions.data as ClusterServiceVersionKind[]),
+    ].find((op) => op.metadata.name === csvName) as ClusterServiceVersionKind;
+    if (typeof tmpCSV !== 'undefined') {
+      const tmpStatus = _.get(tmpCSV, 'status.phase', null);
+      if (tmpStatus !== null) {
+        setInstallObj(tmpCSV);
+        setStatus(tmpStatus);
+        setLoading(false);
+      }
+    } else {
+      // There is no ClusterServiceVersion for the package, so look at Subscriptions/InstallPlans
+      const tmpSub = [...(resources.subscription.data as SubscriptionKind[])].find(
+        (op: SubscriptionKind) =>
+          op.spec.startingCSV === csvName && op.status.state === 'UpgradePending',
+      ) as SubscriptionKind;
+      if (typeof tmpSub !== 'undefined') {
+        const tmpStatus = _.get(tmpSub, 'status.state', null);
+        const ipPlanName = _.get(tmpSub, 'status.installplan.name', '');
+        const tmpInstallPlan: InstallPlanKind = [
+          ...(resources.installPlan.data as InstallPlanKind[]),
+        ].find((ip) => ip.metadata.name === ipPlanName) as InstallPlanKind;
+        setStatus(tmpStatus);
+        if (typeof tmpInstallPlan !== 'undefined') {
+          setInstallObj(tmpInstallPlan);
+          setNeedsApproval(
+            tmpInstallPlan.spec.approval === 'Manual' && tmpInstallPlan.spec.approved === false,
+          );
+          setLoading(false);
+        }
+      }
+    }
+  }, [loading, csvName, namespace, resources]);
+
+  const getIndicator = () => {
```

Comment:
I'd prefer to see some of these helpers pulled out into separate components, rather than so much inline:

```TypeScript
const indicator:React.FC<> = ({prop, prop2}) => {
    if(prop) {
      return <XYZCircleIcon size="lg" />
    }
    if(prop2) {
      return <XYZTriangleIcon size="lg" />
    }
    // etc
}
```

Reply:
I cleaned this up as best I can, but because of how I need to check the status logic, I think it is best I keep it in the component